tests/automation-framework/config/config.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/creasty/defaults"
+)
+
+// LoadConfig reads a config json file and unmarshals it into an object depending on the config object.
+// The functions takes a pointer of the object.
+func LoadConfig(key string, config interface{}) {
+	configPath := os.Getenv("CATTLE_TEST_CONFIG")
+	var configMap map[string]interface{}
+
+	configContents, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		panic(err)
+	}
+
+	err = json.Unmarshal(configContents, &configMap)
```

Comment:
why did we go with json instead of yaml? Json is a subset of yaml so the yaml parser could use json as well.

Reply:
No real reason to be honest, besides it being mentioned in our design doc. Dan was adamant about it for some reason that I cannot remember. But we can do yaml and I can update our design doc.

Reply:
Lets update the design doc.  Unless we find a specific reason not to I think yaml would be more convenvient.  If we find a reason to use json we can always switch to it for the files.

Reply:
For sure I pushed up the changes.